speculatively_test.go

```diff
@@ -201,3 +213,58 @@ func TestContextPropagated(t *testing.T) {
 		t.Errorf("expected Thunk to run %d times, got %d", 2, callCount)
 	}
 }
+
+func TestCompletionCancelsOutstandingThunks(t *testing.T) {
```

Comment:
This new test failed like so on the previous implementation, indicating that we were not properly canceling parallel thunks:

```
--- FAIL: TestCompletionCancelsOutstandingThunks (0.08s)
    speculatively_test.go:237: testThunk: starting call 1
    speculatively_test.go:237: testThunk: starting call 2
    speculatively_test.go:237: testThunk: starting call 3
    speculatively_test.go:240: testThunk: call 3 canceled
    speculatively_test.go:242: unexpected context error in thunk 3: context deadline exceeded
    speculatively_test.go:268: unexpected cancel count: 0 != 2
```